packages/ui/src/components/Accordion/Accordion.tsx

```diff
@@ -9,170 +9,118 @@ import {
-              base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Context](x.5 0 0 x1.25)
-              p-tablet:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Context](0 0 0 0)
-          `,
-            }
-          : {
-              "data-h2-padding": `
-              base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Trigger](x.5 0)
-              base:selectors[>.Accordion__Item > .Accordion__Content](0 x1 x1 x1.25)
-          `,
-            };
-    }
-
-    if (size === "lg") {
-      baseStyles = {
-        "data-h2-height":
-          "base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Icon](x.95)",
-        "data-h2-width":
-          "base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Icon](x.95)",
-        "data-h2-stroke-width":
-          "base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Icon path](1)",
-        "data-h2-font-size":
-          "base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Heading](h5, 1)",
-      };
+const root = tv({
+  base: "group flex flex-col",
+  variants: {
+    mode: {
+      simple: "",
+      card: "overflow-hidden rounded-md bg-white shadow-lg dark:bg-gray-600",
+    },
+    // NOTE: Stub sizes for nested components
+    size: {
+      sm: "",
+      md: "",
+      lg: "",
+    },
+  },
+});
 
-      paddingStyles =
-        mode === "card"
-          ? {
-              "data-h2-padding": `
-              base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Trigger](x1)
-              base:selectors[>.Accordion__Item > .Accordion__Content](0 x1 x1 x2.45)
-            `,
-              "data-h2-margin": `
-              base:selectors[>.Accordion__Item > .Accordion__MetaData](-x.5 0 x1 x2.45)
-              p-tablet:selectors[>.Accordion__Item > .Accordion__MetaData](-x1 0 x1 x2.45)
-              base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Context](x.5 0 0 x1.45)
-              p-tablet:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Context](0 0 0 0)
-            `,
-            }
-          : {
-              "data-h2-padding": `
-          base:selectors[>.Accordion__Item > .Accordion__Header .Accordion__Trigger](x.5 0)
-          base:selectors[>.Accordion__Item > .Accordion__Content](0 x1 x1 x1.45)
-      `,
-            };
-    }
+type AccordionVariants = VariantProps<typeof root>;
+export type AccordionMode = NonNullable<AccordionVariants["mode"]>;
+type RootPrimitiveProps = ComponentPropsWithoutRef<
+  typeof AccordionPrimitive.Root
+>;
 
-    if (mode === "card") {
-      baseStyles = {
-        ...baseStyles,
-        // custom out-of-system colour used for even dark items: Colors/Background/Manual/Dark-30:Foreground-Light-50
-        "data-h2-background-color": `
-          base:selectors[>.Accordion__Item:nth-child(odd)](foreground)
+const AccordionVariantContext = createContext<AccordionVariants>({
```

Comment:
I wonder if this context could eventually be used for an expand-all that doesn't require the container to know the IDs?  :thinking: